From ebf79404fed6248472553f556b3f2548659d0f66 Mon Sep 17 00:00:00 2001
From: ADITHYA S NAIR <adithyasnair2021@gmail.com>
Date: Wed, 31 Jul 2024 20:34:21 +0530
Subject: [PATCH] Changes to made in Client3.py in the getRatio and Main method

Changes to made in Client3.py in the getRatio and Main method
---
 client3.py | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/client3.py b/client3.py
index 3fc09b7..6fad1af 100644
--- a/client3.py
+++ b/client3.py
@@ -35,14 +35,15 @@ def getDataPoint(quote):
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
-    price = bid_price
+    price = (bid_price + ask_price) / 2
     return stock, bid_price, ask_price, price
 
 
 def getRatio(price_a, price_b):
-    """ Get ratio of price_a and price_b """
-    """ ------------- Update this function ------------- """
-    return 1
+    if price_b == 0:
+        # when price_b is 0 avoid throwing ZeroDivisionError
+        return
+    return price_a / price_b
 
 
 # Main
@@ -52,8 +53,10 @@ if __name__ == "__main__":
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
         """ ----------- Update to get the ratio --------------- """
+        prices= {}
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
+            prices[stock] = price
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
-
-        print("Ratio %s" % getRatio(price, price))
+            
+        print("Ratio %s" % (getRatio(prices['ABC'], prices['DEF'])))
-- 
2.37.3.windows.1

